app/main.py

```diff
@@ -1,25 +1,76 @@
+from typing import Any
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: Any, owner: type) -> Any:
+        return getattr(instance, self.privat_name)
+
+    def __set__(self, instance: Any, value: str) -> None:
+        if not isinstance(value, int):
+            raise TypeError
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError
```

Comment:
The `ValueError` raised here should include a message to describe the error, such as 'Value must be between {self.min_amount} and {self.max_amount}'. This will make debugging easier.